test/integration/bastion/bastion_test.go

```diff
@@ -0,0 +1,578 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bastion_test
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"net"
+	"path/filepath"
+	"time"
+
+	awssdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gardener/gardener/extensions/pkg/controller"
+	"github.com/gardener/gardener/pkg/apis/core"
+	"github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/extensions"
+	gardenerutils "github.com/gardener/gardener/pkg/utils"
+	"github.com/gardener/gardener/test/framework"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	awsinstall "github.com/gardener/gardener-extension-provider-aws/pkg/apis/aws/install"
+	awsv1alpha1 "github.com/gardener/gardener-extension-provider-aws/pkg/apis/aws/v1alpha1"
+	"github.com/gardener/gardener-extension-provider-aws/pkg/aws"
+	awsclient "github.com/gardener/gardener-extension-provider-aws/pkg/aws/client"
+	bastionctrl "github.com/gardener/gardener-extension-provider-aws/pkg/controller/bastion"
+	"github.com/gardener/gardener-extension-provider-aws/test/integration"
+)
+
+const (
+	cidrv4              = "172.16.15.14/12" // this is purposefully not normalised
+	cidrv6              = "fd12:3456:789a:1::/64"
+	vpcCIDR             = "10.250.0.0/16"
+	subnetCIDR          = "10.250.0.0/18"
+	publicUtilitySuffix = "public-utility-z0"
+)
+
+var (
+	accessKeyID     = flag.String("access-key-id", "", "AWS access key id")
+	secretAccessKey = flag.String("secret-access-key", "", "AWS secret access key")
+	region          = flag.String("region", "", "AWS region")
+)
+
+func validateFlags() {
+	if len(*accessKeyID) == 0 {
+		panic("need an AWS access key id")
+	}
+	if len(*secretAccessKey) == 0 {
+		panic("need an AWS secret access key")
+	}
+	if len(*region) == 0 {
+		panic("need an AWS region")
+	}
+}
+
+var _ = Describe("Bastion tests", func() {
+	var (
+		ctx = context.Background()
+
+		logger    *logrus.Entry
+		awsClient *awsclient.Client
+
+		testEnv   *envtest.Environment
+		mgrCancel context.CancelFunc
+		c         client.Client
+
+		internalChartsPath string
+	)
+
+	randString, err := randomString()
+	Expect(err).NotTo(HaveOccurred())
+
+	namespaceName := fmt.Sprintf("shoot--%s", randString)
```

Comment:
should the name also have an indicator that it's from a test in case it needs resources leak and need to manually be cleaned up?
@rfranzke how is this done for the other tests that create infrastructure resources? Is there already a naming convention or how is this handled?

Reply:
Maybe something like this could be done as well: https://github.com/gardener/gardener-extension-provider-aws/blob/0d0b8ab4cb62f40116208ca371d3141bbaaf599d/test/integration/infrastructure/infrastructure_test.go#L453-L460

Reply:
Changed it to `aws-bastion-it-[randomstr]`.